src/plugins/move/MoveAction.js

```diff
@@ -0,0 +1,178 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2020, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+export default class MoveAction {
+    constructor(openmct) {
+        this.name = 'Move';
+        this.key = 'move';
+        this.description = 'Move this object from its containing object to another object.';
+        this.cssClass = "icon-move";
+
+        this.openmct = openmct;
+    }
+
+    async invoke(objectPath) {
+        let object = objectPath[0];
+        let oldParent = objectPath[1];
+        let dialogService = this.openmct.$injector.get('dialogService');
+        let dialogForm = this.getDialogForm(object, oldParent);
+        let userInput = await dialogService.getUserInput(dialogForm, { name: object.name });
+
+        this.updateNameCheck(object, userInput.name);
+
+        let parentContext = userInput.location.getCapability('context');
+        let newParent = await this.openmct.objects.get(parentContext.domainObject.id);
+
+        if (this.inNavigationPath(object) && this.openmct.editor.isEditing()) {
+            this.openmct.editor.save();
+        }
+
+        this.addToNewParent(object, newParent);
+        this.removeFromOldParent(oldParent, object);
+
+        if (this.inNavigationPath(object)) {
+            let newObjectPath = await this.openmct.objects.getOriginalPath(object.identifier);
+            let root = await this.openmct.objects.getRoot();
+            let rootChildCount = root.composition.length;
+
+            // if not multiple root children, remove root from path
+            if (rootChildCount < 2) {
+                newObjectPath.pop(); // remove ROOT
+            }
+
+            this.navigateTo(newObjectPath);
+        }
+    }
+
+    updateNameCheck(object, name) {
```

Comment:
Can we update this method name? "updateNameCheck" implies it will return return a boolean and/or is only a check, but this method is mutating the domainObject

Reply:
Yeah good point, I actually just removed the function. It's such simple logic, I just moved it into invoke.